### ISSUE TITLE
singularity fails with default overlay option when running unprivileged

### ISSUE THREAD

**Issue by DrDaveD** [apptainer/singularity#886]
## Version of Singularity:

The current development branch

### Expected behavior

When running in unprivileged mode on a system that allows unprivileged namespaces, I expect the default options to work.

### Actual behavior

With the default "enable overlay = yes", singularity gives the error
```
ERROR  : Could not mount Singularity overlay: Operation not permitted
```
and changing the config value to "enable overlay = no" allows it to work, assuming no other privileged options are requested such as using an image file or binding to a path that is not in the image directory.

### Steps to reproduce behavior

On any system that allows unprivileged singularity (I tried it on EL7.4 and on Debian 9/Stretch each with their appropriate config options), compile from source and install as a non-root user.  Then run 'singularity shell imgdir' where imgdir is a directory with an operating system image.

**Comment by cclerget** [comment]
@DrDaveD actually it's not possible because uid/gid mapping with user namespace only allow current user UID/GID, mapping to root uid/gid is needed for mount. This requires minor change to work

**Comment by DrDaveD** [comment]
@cclerget root uid/gid is not required for mount user namespaces on some systems, as I wrote under "Steps to reproduce behavior", so I don't understand your objection.  All I need to get what I want on those systems is to set 'enable overlay = no'.

**Comment by cclerget** [comment]
@DrDaveD you're right it works only on very recent kernel. It wasn't an objection, I just say it shouldn't be hard to make it works on others systems without loosing benefits of overlayfs

**Comment by DrDaveD** [comment]
Fixed by PR #912
